guide/command-handling/adding-features.md

```diff
@@ -418,17 +418,18 @@ if (!command) return message.channel.send(`There is no command with name or alia
 A lot of library specific structures have `client` as a property. That means you don't have to pass the client reference as a parameter to commands to access for example `client.guilds` or `client.commands`, but can directly access the respective properties directly from the `message` object, as shown in the snippet above.
 :::
 
-Now, in theory, all there is to do, is to delete the previous command from `client.commands` and require the file again. In practice though, you cannot do this that easily as `require()` caches the file. If you were to require it again, you would simply load the previously cached file without any of your changes. In order to remove the file from the cache, you need to add the following line to your code:
+Now, in theory, all there is to do, is to delete the previous command from `client.commands` and require the file again. In practice though, you cannot do this that easily as `require()` caches the file. If you were to require it again, you would simply load the previously cached file without any of your changes. In order to remove the file from the cache, you need to add the following lines to your code:
 
 ```js
-delete require.cache[require.resolve(`./${command.name}.js`)];
+const { id } = Object.values(require.cache).find(module => module.exports === command);
```

Comment:
Will that find there ever return anything? As far as i understand that's comparing raw objects between require's cache and command object taken out from `client.commands` Collection, which would never actually return `true`.

Reply:
I tested it out and it worked

Reply:
Also the exported object of the module is compared

Reply:
From the [Node.js Documentation](https://nodejs.org/api/modules.html#modules_caching "Modules Caching"):
> Modules are cached after the first time they are loaded. This means (among other things) that every call to `require('foo')` will get exactly the same object returned, if it would resolve to the same file.

I guess the one in the cache is also exactly the same.

Reply:
That's not exactly what i asked about. Value in the `client.commands` Collection is not a call to require the command file, but already an object on its own.
What i meant wasn't that objects are different from each other, but that in JavaScript, compring non-primitives (like objects) using `===` doesn't check structure, but if they are distinct from each other, as somewhat explained [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness).

Reply:
In the main file we essentially do `client.commands.set(..., require(filename))` and now we do `client.commands.get(...)`. So `command` is exactly the same object as the one `require()` returned, which is `module.exports`.
Is that right? Or am i missing something and `.set()` and `.get()` don't undo themselves!? Or something else?